packages/hathor-rpc-handler/src/rpcMethods/signWithAddress.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet } from '@hathor/wallet-lib';
+import {
+  TriggerTypes,
+  PinConfirmationPrompt,
+  PinRequestResponse,
+  TriggerHandler,
+  RequestMetadata,
+  SignMessageWithAddressConfirmationPrompt,
+  SignMessageWithAddressConfirmationResponse,
+  SignWithAddressRpcRequest,
+  RpcResponseTypes,
+  SignWithAddressResponse,
+} from '../types';
+import { PromptRejectedError, SignMessageError } from '../errors';
+import { validateNetwork } from '../helpers';
+import { AddressInfoObject } from '@hathor/wallet-lib/lib/wallet/types';
+
+/**
+ * Handles the 'htr_signWithAddress' RPC request by prompting the user for confirmation
+ * and signing the message if confirmed.
+ * 
+ * @param rpcRequest - The RPC request object containing the method and parameters.
+ * @param wallet - The Hathor wallet instance used to sign the message.
+ * @param promptHandler - The function to handle prompting the user for confirmation.
+ *
+ * @returns The signed message if the user confirms.
+ *
+ * @throws {PromptRejectedError} If the user rejects any of the prompts.
+ */
+export async function signWithAddress(
+  rpcRequest: SignWithAddressRpcRequest,
+  wallet: HathorWallet,
+  requestMetadata: RequestMetadata,
+  promptHandler: TriggerHandler,
+) {
+  const { message, addressIndex, network } = rpcRequest.params;
+
+  validateNetwork(wallet, network);
+
+  const base58: string = await wallet.getAddressAtIndex(addressIndex);
+  const addressPath: string = await wallet.getAddressPathForIndex(addressIndex);
```

Comment:
We do have this method in the old facade: https://github.com/HathorNetwork/hathor-wallet-lib/blob/f7403874e7c6aa3d1eb5e401e1e06809cb51a7e5/src/new/wallet.js#L601

Do you mean "new facade"?

If you do mean "new facade", we also don't have nano contracts in it and have a RPC method that interacts with nano contracts

Your idea is to remove just this key so this specific method works in the new facade as well?

Reply:
Yes, I meant new facade, sorry.

I just want to keep consistency, as you said in [this](https://github.com/HathorNetwork/hathor-rpc-lib/pull/2#discussion_r1649470748) comment that you were not going to implement a parameter because the method existed only in the old facade, I was wondering if we should do this here as well.

I prefer that we implement everything that exist in the old facade and create issues to handle (and so we can track) everything that is still not supported by the new facade

Reply:
I created an issue explaining why we won't implement that method for now, you can check it [here](https://github.com/HathorNetwork/hathor-rpc-lib/issues/6)